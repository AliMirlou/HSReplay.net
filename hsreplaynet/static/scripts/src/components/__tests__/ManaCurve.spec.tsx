import ManaCurve from "../ManaCurve";
import * as React from "react";
import renderer from "react-test-renderer";
import { CardObj } from "../../interfaces";

describe("ManaCurve", () => {
	test("renders correctly", () => {
		const firebatDruidDeck: CardObj[] = [
			{ card: { cost: 0 }, count: 2 },
			{ card: { cost: 2 }, count: 2 },
			{ card: { cost: 2 }, count: 2 },
			{ card: { cost: 2 }, count: 2 },
			{ card: { cost: 3 }, count: 2 },
			{ card: { cost: 3 }, count: 2 },
			{ card: { cost: 4 }, count: 2 },
			{ card: { cost: 4 }, count: 2 },
			{ card: { cost: 4 }, count: 2 },
			{ card: { cost: 5 }, count: 1 },
			{ card: { cost: 5 }, count: 2 },
			{ card: { cost: 5 }, count: 1 },
			{ card: { cost: 5 }, count: 1 },
			{ card: { cost: 5 }, count: 2 },
			{ card: { cost: 6 }, count: 1 },
			{ card: { cost: 7 }, count: 2 },
			{ card: { cost: 7 }, count: 1 },
			{ card: { cost: 9 }, count: 1 },
		];

		const component = renderer.create(
			<ManaCurve cards={firebatDruidDeck} />,
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	test("ignores cost differences above 7", () => {
		const threeSevensDeck: CardObj[] = [
			{ card: { cost: 7 }, count: 1 },
			{ card: { cost: 7 }, count: 2 },
		];
		const threeEightsDeck: CardObj[] = [
			{ card: { cost: 8 }, count: 2 },
			{ card: { cost: 8 }, count: 1 },
		];

		const threeSevensComponent = renderer.create(
			<ManaCurve cards={threeSevensDeck} />,
		);

		const threeEightsComponent = renderer.create(
			<ManaCurve cards={threeEightsDeck} />,
		);

		let threeSevensTree = threeSevensComponent.toJSON();
		let threeEightsTree = threeEightsComponent.toJSON();
		expect(threeSevensTree).toMatchObject(threeEightsTree);
	});
});
