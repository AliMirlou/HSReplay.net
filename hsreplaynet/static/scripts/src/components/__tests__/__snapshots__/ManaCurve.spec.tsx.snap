// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManaCurve renders correctly 1`] = `
<ul
  className="mana-curve"
>
  <li>
    <span
      data-cost={0}
      data-count={2}
      style={
        Object {
          "height": "28.571428571428573%",
        }
      }
    />
  </li>
  <li>
    <span
      data-cost={1}
      data-count=""
      style={
        Object {
          "height": "0%",
        }
      }
    />
  </li>
  <li>
    <span
      data-cost={2}
      data-count={6}
      style={
        Object {
          "height": "85.71428571428571%",
        }
      }
    />
  </li>
  <li>
    <span
      data-cost={3}
      data-count={4}
      style={
        Object {
          "height": "57.142857142857146%",
        }
      }
    />
  </li>
  <li>
    <span
      data-cost={4}
      data-count={6}
      style={
        Object {
          "height": "85.71428571428571%",
        }
      }
    />
  </li>
  <li>
    <span
      data-cost={5}
      data-count={7}
      style={
        Object {
          "height": "100%",
        }
      }
    />
  </li>
  <li>
    <span
      data-cost={6}
      data-count={1}
      style={
        Object {
          "height": "14.285714285714286%",
        }
      }
    />
  </li>
  <li>
    <span
      data-cost="7+"
      data-count={4}
      style={
        Object {
          "height": "57.142857142857146%",
        }
      }
    />
  </li>
</ul>
`;
